lxd/instance/instance_utils.go

```diff
@@ -128,14 +128,14 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, expanded bool, instanc
 }
 
 func validConfigKey(os *sys.OS, key string, value string, instanceType instancetype.Type) error {
+	// Disallow keys with container-specific prefixes such as "linux.sysctl." and "limits.kernel." for VMs.
+	if instanceType == instancetype.VM && shared.StringHasPrefix(key, instancetype.ConfigKeyPrefixesContainer...) {
+		return fmt.Errorf("%q isn't supported for %q", key, instanceType)
+	}
+
 	// Check if the key is a valid prefix and whether or not it requires a subkey.
 	knownPrefixes := append(instancetype.ConfigKeyPrefixesAny, instancetype.ConfigKeyPrefixesContainer...)
 	if strings.HasSuffix(key, ".") {
```

Comment:
Shouldn't we also reject anything that ends with a trailing `.` as that's never going to be a valid config key anyway so why bother doing more checks?

Reply:
We can do that, however, the additional check allows us to throw more specific error messages:
`"Unknown configuration key: %q"` if the key is not known and `"%q requires a subkey"` if the key is known but missing a subkey. Also, the check is pretty inexpensive computationally since the `knownPrefixes` slice is only 5 elements.